examples/scripts/denoise_cplx_tv_nlpadmm.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This file is part of the SCICO package. Details of the copyright
+# and user license can be found in the 'LICENSE.txt' file distributed
+# with the package.
+
+r"""
+Complex Total Variation Denoising with NLPADMM Solver
+=====================================================
+
+This example demonstrates solution of a problem of the form
+
+$$\argmin_{\mathbf{x}} \; f(\mathbf{x}) + g(\mathbf{z}) \; \text{such
+ that}\; H(\mb{x}, \mb{z}) = 0 \;,$$
+
+where $H$ is a function, via a variant of the proximal ADMM
+algorithm for problems with a non-linear operator constraint
+:cite:`benning-2016-preconditioned`. The example problem represents
+total variation (TV) denoising applied to a complex image with
+piece-wise smooth magnitude and non-smooth phase. The appropriate TV
+denoising formulation for this problem is
+
+$$\mathrm{argmin}_{\mathbf{x}} \; (1/2) \| \mathbf{y} - \mathbf{x}
+\|_2^2 + \lambda \| C(\mathbf{x}) \|_{2,1} \;,$$
+
+where $\mathbf{y}$ is the measurement, $\|\cdot\|_{2,1}$ is the
+$\ell_{2,1}$ mixed norm, and $C$ is a non-linear operator that applies
+a linear difference operator to the magnitude of a complex array.
```

Comment:
The operator applies an operator. Sounds odd.

Reply:
Rephrased.